### PR TITLE
[PyOV][Docs] Update docs with new way of using properties

### PR DESCRIPTION
**Pull request by akuporos**
### Details:
 - follow up of the https://github.com/openvinotoolkit/openvino/pull/20007
### Tickets:
 - *ticket-id*

### CODE REVIEW

File: docs/snippets/ov_properties_api.py [hunk 1]
```diff
@@ -17,48 +17,48 @@ def main():
...
-    nthreads = compiled_model.get_property(props.inference_num_threads())
+    nthreads = compiled_model.get_property(props.inference_num_threads)
     # [inference_num_threads]
 
     if "GPU" not in available_devices:
         return 0
 
     # [multi_device]
-    config = {device.priorities(): "CPU,GPU"}
+    config = {device.priorities: "CPU,GPU"}
```
Comment (ilya-lavrenov) [inline docs/snippets/ov_properties_api.py:60]: Why don't we use:
```python
device.priorities("CPU,GPU")
```
as in C++?

Comment (jiwaszki) [reply docs/snippets/ov_properties_api.py:60]: @ilya-lavrenov longer answer as I have to explain design choice.

**tl;dr this is a design choice and there is no overload in Python API to handle "variadic arguments".**

As C++ allows variadic arguments, in Python these are called "arbitrary arguments" `*args` and `**kwargs` that allow to dispatch "the rest" of not defined arguments based on position (args) or argument name (kwargs).
One example of using such features is:
https://numpy.org/doc/stable/reference/generated/numpy.apply_along_axis.html
```python
In [8]: def ov_func(a, e, f):
   ...:     return (a[0] + a[1]) + e - f
   ...: b = np.array([[1, 2], [3, 4], [5, 6]])
   ...: np.apply_along_axis(ov_func, 1, b, 2, 1)
Out[8]: array([ 4,  8, 12])
```
... and examples of improper usage:
```python
In [15]: np.apply_along_axis(ov_func, 1, b, e=2, 1)  # using only one keyword
  Cell In[15], line 1
    np.apply_along_axis(ov_func, 1, b, e=2, 1)
                                             ^
SyntaxError: positional argument follows keyword argument

In [16]: np.apply_along_axis(ov_func, 1, b, f=2, e=1)  # wrong assigning by accident 
Out[16]: array([ 2,  6, 10])
```

Now taking Python and C++ under consideration:
* Some takes `AnyMap` as argument (note: map of pairs `std::map<std::string, Any>`) which is similar to Python's `dict`.
That's why `config=...` argument in Python API feels OK, map is a dict -- end of the story. This is **error-prone**, **easy to extend/work** with different `compile_model` signatures (i.e. if any changes appear in future: new argument between variadics and `device_name` is added) and **works on well-known dict-type**. Especially editing existing configs is easier this way. https://github.com/openvinotoolkit/openvino/blob/6e1e8105fe8feda4f4bf272355c33a0f9fd748e5/src/inference/include/openvino/runtime/core.hpp#L163-L165
```python
# You can always rely on conversions from tuple to dict:
tmp0 = ov.properties.hint.inference_precision(ov.Type.f32)  # creates a Tuple[str, OVAny]
tmp1 = ...
...
my_config = dict([tmp0, tmp1, ...])  # produces: {'INFERENCE_PRECISION_HINT': <OVAny class>, ...}
# Or simply use new more friendly interface of:
my_config = {
    ov.properties.hint.inference_precision: ov.Type.f32,
    # ov.properties.cache_dir: "some/path/here/",  # (un-)commenting is really easy, this is only one place
}
# Later on simply use config=... to ensure any overload path, config stays in place:
compiled = core.compile_model(model, my_config)
compiled = core.compile_model(model, config=my_config)
compiled = core.compile_model(model, "CPU", config=my_config)
compiled = core.compile_model(model, device_name="CPU", config=my_config)
```
* Afaik there is no simple way to bind variadic templates in pybind itself. It requires overload for each Property class being added. Ref: https://pybind11.readthedocs.io/en/stable/advanced/functions.html#binding-functions-with-template-parameters
Solution with `py::args` is not guaranteed to work as well. As you need to know number of args at the compile-time.
* Even if above scenario somehow works, variadic arguments are still converted to `AnyMap` later:
https://github.com/openvinotoolkit/openvino/blob/6e1e8105fe8feda4f4bf272355c33a0f9fd748e5/src/inference/include/openvino/runtime/core.hpp#L179-L185
Here the story would be a lot different. A lot of impl details to write and overall cost of up-keeping/extending of it is significant.
```python
# Assume properties:
tmp0 = ov.properties.hint.inference_precision(ov.Type.f32)
tmp1 = ...
# # Examples with explanations:
# Which overload will pybind take? Is tmp0 a tuple or device_name? Maybe there is device_name in-between?
# When using *args our dispatcher must iterate over them to determine which arguments are Properties.
# Even discarding the cost of it, currently it is still needed to perform `py_object_to_any`.
# Is adding complexity worth it?
compiled = core.compile_model(model, tmp0, tmp1)
compiled = core.compile_model(model, "CPU", tmp0, tmp1)
# Some cases where extra handling will be needed, again extra complexity.
# All will simply break, so it needs to handle this situation (with error) or simply switch to config with dict.
compiled = core.compile_model(model, {here_is_the_config}, tmp1)
compiled = core.compile_model(model, config=..., tmp1)
compiled = core.compile_model(model, device_name="CPU", tmp0, tmp1)
```

So summing it up, is it worth it to imitate C++ API 1:1? In this case I don't want to sacrifice a really great Python interface. On top of that, `*args` and `**kwargs` are usually used when there is nothing better to replace them (i.e. gluing arguments together like in `np.apply_along_axis`). Of course I am speaking only from my own experience.

I am open to discussion but I think some language-specific behaviors and ideas should be kept.

**Review by jiwaszki — APPROVED** [review]

File: docs/snippets/ov_properties_api.py (hunk 1 above)
Comment (mlukasze) [reply docs/snippets/ov_properties_api.py:60]: is it possible to add a method `properties()` beside of property `properties`?
in that case we would have both goals achieved:

* `properties()` methods would mimic CPP behavior and make it easier for CPP developers to understand
* but in same time `properties` approach will be more `pythonic` and for those who used to that kind of setting parameters, would be easier to understand

**Review by mlukasze — APPROVED** [review]
